core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -370,7 +371,7 @@ public String toString() {
     toStringHelper.add("affinity", affinity);
     toStringHelper.add("executor", executor != null ? executor.getClass() : null);
     toStringHelper.add("compressorName", compressorName);
-    toStringHelper.add("customOptions", Arrays.toString(customOptions));
+    toStringHelper.add("customOptions", Arrays.deepToString(customOptions));
```

Comment:
@dapengzhang0 I opened a PR that changes `withOption` accordingly: https://github.com/grpc/grpc-java/pull/1919